File: js/examples/app-and-signals.ts
// App bridge + notifications + signal levels — drive the terminal UI from a widget.
// Run:  npx tsx examples/app-and-signals.ts
import { ColibriClient } from "../src/index.js";

const client = await ColibriClient.discover();
const [EXCHANGE, SYMBOL] = ["BinanceSpot", "BTCUSDT"];

// openSymbol — open ONE coin + surface the window ("see the move → open the book").
// A POST /app/panels {activate:true} under the hood — returns the created slot.
console.log("open-symbol:", await client.openSymbol(EXCHANGE, SYMBOL));

// POST /app/combos  — fan the coin across every connection that lists it
console.log("combo:", await client.openCombo(SYMBOL, "window"));

// POST /notifications  — raise a toast in the terminal
console.log("notify:", await client.notify("Hello from a widget 👋", "info", "my-widget"));

// POST /signals  — post a free-text market signal into Notifications → API tab
console.log("signal:", await client.signal(EXCHANGE, SYMBOL, "whale wall pulled"));

// Signal levels — API-owned price alerts, ALSO drawn on the ladder.
const last = Number((await client.book(EXCHANGE, SYMBOL)).lastPrice);

// POST /signal-levels  → 201. A level fires AT MOST ONCE: oneShot removes it on fire, otherwise
// it is kept marked isTriggered (sweep those with deleteTriggeredSignalLevels()).
const level = await client.createSignalLevel({
  exchange: EXCHANGE,
  symbol: SYMBOL,
  price: (last * 1.02).toFixed(2), // fire when price crosses +2%
  direction: "above",
  note: "breakout watch",
  oneShot: true,
});
console.log("\ncreated level:", level.id, "@", level.price, "triggered:", level.isTriggered);

// GET /signal-levels?exchange=&symbol=  (a connectionId filter is available too)
console.log("levels now:", (await client.signalLevels(EXCHANGE, SYMBOL)).map((l) => `${l.price} (${l.direction})`));

// DELETE /signal-levels/{id}
console.log("deleted:", await client.deleteSignalLevel(level.id));

// DELETE /signal-levels/triggered  — sweep every already-fired level
console.log("swept fired levels:", await client.deleteTriggeredSignalLevels());

File: python/examples/app_and_signals.py
"""App bridge + notifications + signal levels — drive the terminal from a widget.

    python examples/app_and_signals.py
"""
from colibri import ColibriClient

client = ColibriClient.discover()
EXCHANGE, SYMBOL = "BinanceSpot", "BTCUSDT"

# open_symbol — open one coin + surface the window (a POST /app/panels {activate: true} under the hood)
print("open-symbol:", client.open_symbol(EXCHANGE, SYMBOL))

# POST /app/combos  — fan across every connection that lists it
print("combo:", client.open_combo(SYMBOL, target="window"))

# POST /notifications  — raise a toast
print("notify:", client.notify("Hello from a widget", severity="info", source="my-widget"))

# POST /signals  — free-text market signal into Notifications -> API tab
print("signal:", client.signal(EXCHANGE, SYMBOL, "whale wall pulled"))

# Signal levels — API-owned price alerts, also drawn on the ladder.
last = float(client.book(EXCHANGE, SYMBOL)["lastPrice"])

# POST /signal-levels  -> 201. A level fires AT MOST ONCE: one_shot removes it on fire, otherwise
# it is kept marked isTriggered (sweep those with delete_triggered_signal_levels()).
level = client.create_signal_level(EXCHANGE, SYMBOL, price=f"{last * 1.02:.2f}", direction="above", note="breakout watch", one_shot=True)
print("\ncreated level:", level["id"], "@", level["price"], "triggered:", level["isTriggered"])

# GET /signal-levels?exchange=&symbol=  (a connectionId filter is available too)
print("levels now:", [f"{l['price']} ({l['direction']})" for l in client.signal_levels(EXCHANGE, SYMBOL)])

# DELETE /signal-levels/{id}
print("deleted:", client.delete_signal_level(level["id"]))

# DELETE /signal-levels/triggered  — sweep every already-fired level
print("swept fired levels:", client.delete_triggered_signal_levels())
